onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -602,7 +602,7 @@ def bind_input(self, name, device_type, device_id, element_type, shape, buffer_p
         :param name: input name
         :param device_type: e.g. cpu, cuda, cann
         :param device_id: device id, e.g. 0
-        :param element_type: input element type
+        :param element_type: input element type. It can be either numpy type (like numpy.float32) or an integer for onnx type (like onnx.TensorProto.BFLOAT16)
```

Comment:
In python code below we can simply test for type. How do we tell the difference in this API?

Reply:
In C/C++, we can have function overloading like print(int) and print(float), as long as compiler can distinguish the signature, it is fine to use same API name.

Pybind11 supports function overloading directly by allowing multiple m.def() calls with different static cast signatures for each function. It routes the call to the correct overloaded function based on the argument types passed from Python.

BTW, we have unit tests to cover both types in onnxruntime_test_python_iobinding.py. That means numpy type and onnx type are routed to different C/C++ code correctly.